bubblesortby.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+def bubblesort_by(array)
+  result = array
+  loop do
+    swap = 0
+    result.each_with_index do |_, i|
+      break if result[i + 1].nil?
+
+      if yield(result[i], result[i + 1]).positive?
+        result[i], result[i + 1] = result[i + 1], result[i]
+        swap += 1
+      end
+    end
+    break if swap.zero?
+  end
+  puts result
```

Comment:
The testing is failing. Please remove the puts and leave result.

Reply:
We removed the puts in the function as demanded!